code/modules/vore/eating/belly_vr.dm

```diff
@@ -303,22 +302,12 @@
 					brain.forceMove(owner)
 					items_preserved += brain
 					internal_contents += brain
-			if(W == H.get_equipped_item(slot_wear_id))
-				H.remove_from_mob(W,owner)
-				internal_contents += W
-			if(W == H.get_equipped_item(slot_w_uniform))
-				var/list/stash = list(slot_r_store,slot_l_store,slot_wear_id,slot_belt)
-				for(var/stashslot in stash)
-					var/obj/item/SL = H.get_equipped_item(stashslot)
-					if(SL)
-						SL.forceMove(owner)
-						internal_contents += SL
-				H.remove_from_mob(W,owner)
-				internal_contents += W
-			else
-				if(!(istype(W,/obj/item/organ) || istype(W,/obj/item/weapon/storage/internal) || istype(W,/obj/screen)))//Don't drop organs or pocket spaces
-					H.remove_from_mob(W,owner)
-					internal_contents += W
+			for(var/slot in slots)
+				var/obj/item/thingy = M.get_equipped_item(slot = slot)
+				if(thingy)
+					M.unEquip(thingy,force = TRUE)
+					thingy.forceMove(owner)
+					internal_contents |= thingy
```

Comment:
This will make any items that are removed as a result of removing the intended item fall onto the floor below the person. Pockets/belt/ID from uniform, suit storage from suits, etc.

Reply:
Uniform is one of the last things in that list, and as such it doesn't get removed. Uniform is the only object I'm aware of that causes you to throw all your stuff everywhere, and said stuff has already been removed at that point.

I did test it, and it works fine. I digested several people with full gear on and nothing falls on the floor (except for one time I digested someone with a ton of brute damage (100) and their feet came off and so their shoes fell off 👀 BUT THAT ASIDE...)